### PR TITLE
RepeatEndStep has the same treatment as EndStep under ComputerAwareStep

### PR DESCRIPTION
**Pull request by rdtr**
RepeatEndStep has the same treatment as EndStep under ComputerAwareStep

### CODE REVIEW

**Review by saikiranboga — APPROVED** [review]

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/IdentityRemovalStrategy.java [hunk 1]
```diff
@@ -65,8 +66,12 @@ public void apply(final Traversal.Admin<?, ?> traversal) {
                 // gets to be:
                 // [UnionStep([[VertexStep(OUT,vertex), EndStep], [IdentityStep, EndStep], [VertexStep(OUT,vertex), EndStep]])]
                 // EndStep seems to just behave like a identity() in the above case, but perhaps it is more consistent
-                // to keep the identity() placeholder rather than a step that doesn't actually exist
-                if (!(identityStep.getNextStep() instanceof ComputerAwareStep.EndStep && traversal.getSteps().size() == 2)) {
+                // to keep the identity() placeholder rather than a step that doesn't actually exist.
```
Comment (Cole-Greer) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/IdentityRemovalStrategy.java:69]: Nit: Would you mind making a quick update to the javadoc comment at the top of this class to mention the EndStep and RepeatEndStep exceptions? That comment looks a bit out of date.

**Comment by Cole-Greer** [comment]
LGTM VOTE +1

**Comment by xiazcy** [comment]
VOTE + 1

**Comment by kenhuuu** [comment]
VOTE +1

**Comment by rdtr** [comment]
Looks like ready to merge ?

**Pull request merged**